src/spikeinterface/core/base.py

```diff
@@ -836,41 +836,62 @@ def _check_if_dumpable(d):
         return d["dumpable"]
 
 
-def _load_extractor_from_dict(dic):
-    cls = None
+def _load_extractor_from_dict(dic) -> BaseExtractor:
+    """
+    Convert a dictionary into an instance of BaseExtractor or its subclass.
+
+    This function takes a dictionary that represents the state of an extractor,
+    and reconstructs the extractor from the dictionary.
+
+    Parameters
+    ----------
+    dic : dict
+        A dictionary representation of the extractor which  must contain the following keys:
+        - "kwargs": A dictionary of keyword arguments to be passed to the extractor class upon instantiation.
+        - "class": The full name of the extractor class, including the module name.
+        - "version": The version of the extractor class used when the dictionary was created.
+        - "annotations": A dictionary of annotations.
+        - "properties": A dictionary of properties.
+
+    Returns
+    -------
+    BaseExtractor
+        An instance of BaseExtractor or its subclass.
+    """
+    extractor_class = None
     class_name = None
 
     if "kwargs" not in dic:
         raise Exception(f"This dict cannot be load into extractor {dic}")
-    kwargs = deepcopy(dic["kwargs"])
```

Comment:
This deepcopy was somhow important and not important.

For instance if you want to clone a recording in order to make an inplace change (on properties or probe or whatever). The two unerlying data (list, dict, arrays) must be copied.

I woulda dd an option for this and then check in detail in with case a copy needed or not.

Reply:
In short we should have a copy but not recurssive.

Reply:
There are some problems with `deepcopy`

When I use it in the Lussac tests, sometimes the annotations dissapear ...
So I wouldn't completely trust it!

Reply:
I think the solution above re-creates `list` and `dict` right?

Maybe we can add `if not isinstance(obj, BaseExtractor): deepcopy(obj)`, since the problem was deep-copying multiple times the extractors.

Reply:
That said, as @DradeAW mentions above this recreates the kwargs. I actually fail to see how this changes will fail to support the desired use-case. The kwargs are re-instantiated to initialize a class and then become _kwargs of the new class which should not be changed in there.

 The properties were taken directly from `dict` in the previous implementation (the one in main) and that's left as it is (see line 894) so I don't think this use case was ever supported.